### ISSUE TITLE
IForEachLoopOperation.LoopControlVariable is null for some error scenarios

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#23820]
```
        var ar = new int[] { 1, 2, 3 };

        foreach(ref readonly v in ar)
        {
        }
```

Affected unit-tests:
```
Microsoft.CodeAnalysis.CSharp.UnitTests.Parsing.RefReadonlyTests.RefReadOnlyForEachAreDisallowed
Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen.CodeGenDeconstructTests.MixedDeconstruction_06
Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen.CodeGenDeconstructTests.ShortDiscardDisallowedInForeach
Microsoft.CodeAnalysis.CSharp.UnitTests.IOperationTests.IForEachLoopStatement_WithInvalidLoopControlVariable
```

**Comment by AlekseyTs** [comment]
A workaround should be removed from 
Microsoft.CodeAnalysis.Test.Utilities.TestOperationVisitor.VisitForEachLoop once this issue is addressed.

**Comment by AlekseyTs** [comment]
CC @dotnet/analyzer-ioperation

**Comment by 333fred** [comment]
Fixed in #25848.
